shared/src/commonTest/kotlin/co/nimblehq/avishek/phong/kmmic/data.repository/SurveyRepositoryTest.kt

```diff
@@ -142,4 +142,31 @@ class SurveyRepositoryTest {
                 this.awaitError().message shouldBe MockUtil.mockThrowable.message
             }
         }
+
+    @Test
+    fun `when get detail survey is succeeded, it returns survey`() =
+        runTest {
+            given(mockSurveyRemoteDataSource)
+                .function(mockSurveyRemoteDataSource::getSurvey)
+                .whenInvokedWith(any())
+                .thenReturn(flow { emit(MockUtil.mockSurveyApiModel) })
+
+            repository.getSurvey("id").test {
+                this.awaitItem() shouldBe MockUtil.mockSurveyApiModel.toSurvey()
+                this.awaitComplete()
+            }
```

Comment:
Optional:
```suggestion
        runTest {
            given(mockSurveyRemoteDataSource)
                .function(mockSurveyRemoteDataSource::getSurvey)
                .whenInvokedWith(any())
                .thenReturn(flowOf(MockUtil.mockSurveyApiModel))

            repository.getSurvey("id").first() shouldBe MockUtil.mockSurveyApiModel.toSurvey()
        }
```

Reply:
solved in [05a48e0](https://github.com/nimblehq/avishek-phong-kmm-ic/pull/119/commits/05a48e0eab1776cafd561a4724b5775f2122792c)